src/SupportInfillPart.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2017 Ultimaker B.V.
+// CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include "SupportInfillPart.h"
+#include "support.h"
+
+using namespace cura;
+
+
+SupportInfillPart::SupportInfillPart(const Polygons& outline, coord_t support_line_width, int infill_overlap, int inset_count_to_generate)
+    : outline(outline)
+    , support_line_width(support_line_width)
+    , infill_overlap(infill_overlap)
+    , inset_count_to_generate(inset_count_to_generate)
+{
+    insets.clear();
+    infill_area.clear();
+    infill_areas_per_combine_per_density.clear();
+}
+
+
+SupportInfillPart::~SupportInfillPart()
+{
+    support_line_width = 0;
+    infill_overlap = 0;
+    inset_count_to_generate = 0;
+    outline.clear();
+    insets.clear();
+    infill_area.clear();
+    infill_areas_per_combine_per_density.clear();
+}
+
+
+bool SupportInfillPart::generateInsetsAndInfillAreas()
+{
+    // generate insets, use the first inset as the wall line, and the second as the infill area
+    AreaSupport::generateOutlineInsets(
+        this->insets,
+        this->outline,
+        this->inset_count_to_generate,
+        this->support_line_width);
+    if (this->inset_count_to_generate > 0 && this->insets.empty())
+    {
+        return false;
+    }
+
+    // get infill area
+    if (this->inset_count_to_generate == 0)
```

Comment:
This makes it clear that it is a member variable. Sometime some functions have too many arguments and local variables which makes it hard to know if something is a member variable.